cockatrice/src/filter_string.cpp

```diff
@@ -0,0 +1,217 @@
+            auto op = sv[0].get<QString>();
+            return [=](int s) {
+                if (op == ">")
+                    return s > arg;
+                if (op == ">=")
+                    return s >= arg;
+                if (op == "<")
+                    return s < arg;
+                if (op == "<=")
+                    return s <= arg;
+                if (op == "=")
+                    return s == arg;
+                if (op == ":")
+                    return s == arg;
+                if (op == "!=")
+                    return s != arg;
+                return false;
+            };
+        };
+
+        search["NumericValue"] = [](const peg::SemanticValues &sv) -> int {
+            return QString::fromStdString(sv.str()).toInt();
+        };
+
+        search["NumericOperator"] = [](const peg::SemanticValues &sv) -> QString {
+            return QString::fromStdString(sv.str());
+        };
+
+        search["RegexString"] = [](const peg::SemanticValues &sv) -> StringMatcher {
+            auto target = sv[0].get<QString>();
+            return [=](const QString &s) { return s.QString::contains(target, Qt::CaseInsensitive); };
+        };
+
+        search["OracleQuery"] = [](const peg::SemanticValues &sv) -> Filter {
+            StringMatcher matcher = sv[0].get<StringMatcher>();
+            return [=](CardData x) { return matcher(x->getText()); };
+        };
+
+        search["ColorQuery"] = [](const peg::SemanticValues &sv) -> Filter {
+            char target = sv[0].get<char>();
+            return [=](CardData x) {
+                if (target == 'm')
+                    return x->getColors().length() > 1;
+                return x->getColors().contains(target);
+            };
+        };
+
+        search["CMCQuery"] = [](const peg::SemanticValues &sv) -> Filter {
+            NumberMatcher matcher = sv[0].get<NumberMatcher>();
+            return [=](CardData x) -> bool { return matcher(x->getProperty("cmc").toInt()); };
+        };
+        search["PowerQuery"] = [](const peg::SemanticValues &sv) -> Filter {
+            NumberMatcher matcher = sv[0].get<NumberMatcher>();
+            return [=](CardData x) -> bool { return matcher(x->getPowTough().split("/")[0].toInt()); };
+        };
+        search["ToughnessQuery"] = [](const peg::SemanticValues &sv) -> Filter {
+            NumberMatcher matcher = sv[0].get<NumberMatcher>();
+            return [=](CardData x) -> bool {
+                auto parts = x->getPowTough().split("/");
+                return matcher(parts.length() == 2 ? parts[1].toInt() : 0);
+            };
+        };
+
+        search["GenericQuery"] = [](const peg::SemanticValues &sv) -> Filter {
+            StringMatcher matcher = sv[0].get<StringMatcher>();
+            return [=](CardData x) { return matcher(x->getName()); };
+        };
+
+        search["Color"] = [](const peg::SemanticValues &sv) -> char {
+            switch (sv.choice()) {
+                case 0:
```

Comment:
Should we make a small `const map {W,U,B,R,G, }` and just return the index instead of the case switches?

Reply:
Well you know what an array of chars is ;)